internal/service/model_service.go

```
@@ -535,6 +535,27 @@ type CreateInstanceModelInfo struct {
 
+func validateBedrockAPIKeyAuth(providerName, apiKey string) (bool, error) {
+	if !strings.EqualFold(providerName, "Bedrock") {
+		return false, nil
+	}
+	var config struct {
+		AuthMode string `json:"auth_mode"`
+		APIKey   string `json:"bedrock_api_key"`
+		Region   string `json:"bedrock_region"`
+	}
+	if json.Unmarshal([]byte(apiKey), &config) != nil || config.AuthMode != "bedrock_api_key" {
+		return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 547) but it returns nil

(nilerr)